### DISCUSSION TITLE
Alternatives to PCA

### DISCUSSION THREAD

**Discussion by dagarfield** [brianhie/geosketch#15] [Q&A]
Heya!

Big fan of the tool, which I've been using on large projects with many samples. Up until now, I've gotten around batch-effect issues by feeding the program PCA corrected with fastMNN. Seems to work quite well, but I have a few cases where I am worried (perhaps irrelevantly?) about larger batch effects. I'm thus wondering if you had any thoughts on feeding the method other embeddings such as latent spaces calculated with scVI or similar. 

Thanks!

**Comment by brianhie** [comment]
In theory you could apply geosketch to any other embedding -- however, an advantage of geosketch on SVD dimensionality reduced embeddings is that we apply geosketch in the order of the largest singular values, and geosketch uses equal-width bins across all dimensions, so naturally geosketch will effectively ignore many of the dimensions corresponding to the smallest singular values. You could very easily compute SVD on the scVI embeddings and get the same property. Also, we haven't tested geosketch on any other embeddings beyond SVD. Let me know if that doesn't make sense!

**Reply by dagarfield** [reply]
Totally. Thanks!
